### ISSUE TITLE
Script random Stop

### ISSUE THREAD

**Issue by basalisco** [TheNetsky/Microsoft-Rewards-Script#54]
I don't know why but when the script is run on windows it doesn't give any problems. But when run on linux in headless mode, I often find the log stopped without any errors but it didn't move forward.

```
[26/1/2024, 17:03:38] [PID: 29708] [LOG] [BROWSER] Created browser with User-Agent: "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36 Edg/119.0.0.0"
[26/1/2024, 17:03:38] [PID: 29708] [LOG] [MAIN] Starting DESKTOP browser
[26/1/2024, 17:03:59] [PID: 29708] [LOG] [LOGIN] Email entered successfully
[26/1/2024, 17:04:02] [PID: 29708] [LOG] [LOGIN] Password entered successfully

```

AND STOP

**Comment by TheNetsky** [comment]
I think there's an issue with playwright n headless, am not sure.

**Comment by basalisco** [comment]
> I think there's an issue with playwright n headless, am not sure.

I have done some tests and it will be linked to the moment in which the browser asks if you want to save the session so as not to log in again, at that moment everything remains still. I saved the session by starting it in non-headless mode, activated the fingerprint (true) option headless mode reactivated, and everything seems to be going smoothly at the moment.

**Comment by TheNetsky** [comment]
And you do not have 2FA?

**Comment by basalisco** [comment]
> And you do not have 2FA?

No.

**Comment by mgrimace** [comment]
Try `npx playwright install` after `npm i` step - I was running into a similar problem on a headless linux homeserver setup. I forget where I caught it but at some point during the install/build of the script there was a subtle prompt to run that command that (for me) was easily missed.

**Comment by basalisco** [comment]
> Try `npx playwright install` after `npm i` step - I was running into a similar problem on a headless linux homeserver setup. I forget where I caught it but at some point during the install/build of the script there was a subtle prompt to run that command that (for me) was easily missed.

Hello, I also tried to reinstall everything, I have the impression that the problem is due to the page where it asks "do you want to stay logged in?". At the moment I have ''solved'' by starting the script on another machine with desktop, enabling fingerprint and copying the contents of the session folder to my server where the script runs in headless mode

**Comment by mgrimace** [comment]
Ok, testing this on a fresh install, headless, debian 12 and experiencing the same issue.
Installed latest version of the script, latest NPM, and used both `npx playwright install` and `npx playwright install-deps` as prompted. Built the script, then ran. Stops at `[LOGIN] Password entered successfully` - it doesn't appear frozen, just doesn't progress at all. 

No 2FA, working as expected on 1.4.2.

Additional info:

Debian LXC (proxmox), 2CPU, 2gb Ram, 8gb ssd (script sitting around 40% ram 3%cpu while stuck at this step; not frozen, able to SSH, etc.)
Config is as below, 2 local accounts, headless, geo enabled, search delay changed d/t 15 minute search timeout in Canada:

```yaml
{
    "baseURL": "https://rewards.bing.com",
    "sessionPath": "sessions",
    "headless": true,
    "runOnZeroPoints": false,
    "clusters": 2,
    "saveFingerprint": false,
    "workers": {
        "doDailySet": true,
        "doMorePromotions": true,
        "doPunchCards": true,
        "doDesktopSearch": true,
        "doMobileSearch": true
    },
    "searchSettings": {
        "useGeoLocaleQueries": true,
        "scrollRandomResults": true,
        "clickRandomResults": true,
        "searchDelay": {
            "min": 300000,
            "max": 900000
        },
        "retryMobileSearch": true
    },
    "webhook": {
        "enabled": false,
        "url": ""
    }
}
```
Not sure what else would be helpful, it isn't 'crashing' or freezing, just seems to hang indefinitely here as far as I can tell

**Comment by TheNetsky** [comment]
Fixed in 1.4.4
